routing/message/handlers/state/shadow_state_handler_test.go

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) 2023 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+package state
+
+import (
+	"testing"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/eclipse-kanto/aws-connector/config"
+	"github.com/eclipse-kanto/aws-connector/routing/message/handlers"
+	"github.com/eclipse-kanto/aws-connector/routing/message/handlers/passthrough"
+	"github.com/eclipse-kanto/suite-connector/connector"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCreateDefaultHandler(t *testing.T) {
+	handler := CreateDefaultShadowStateHandler()
+
+	require.NoError(t, handler.Init(settings(), watermill.NopLogger{}))
+	assert.Equal(t, "shadow_state_handler", handler.Name())
+	assert.Equal(t, "$aws/things/test:device/shadow/update/accepted,$aws/things/test:device/shadow/delete/accepted,$aws/things/test:device/shadow/name/+/update/accepted,$aws/things/test:device/shadow/name/+/delete/accepted", handler.Topics())
+}
+
+func TestErrorWhenTopicMissingInMessage(t *testing.T) {
+	handler := CreateDefaultShadowStateHandler()
+	handler.Init(settings(), watermill.NopLogger{})
+	payload := `{
+		"state": {
+			"reported": {
+				"test": "value"
+			}
+		}
+	}`
+
+	message := &message.Message{Payload: []byte(payload)}
+
+	result, err := handler.HandleMessage(message)
+	assert.NotNil(t, err)
+	assert.Nil(t, result)
+}
+
+func TestErrorWhenUpdatingAndPayloadNotMap(t *testing.T) {
```

Comment:
This + the next 3 tests can be combined in one, using a for loop. The other tests could be refactored as well.